core/modules/dashboard/includes/block.welcome.inc

```diff
@@ -68,18 +81,29 @@ class DashboardWelcomeBlock extends Block {
       );
     }
 
-    $next_steps_links = array();
-    $next_steps_links['node/add/post'] = t('Create a new post');
+    // Add the second column of links.
+    $links_col2 = array();
     $about_node = node_load(2);
     if ($about_node) {
-      $next_steps_links['node/2/edit'] = t('Edit the %title page', array('%title' => $about_node->title));
+      $links_col2['node/2/edit'] = t('Edit the %title page', array('%title' => $about_node->title));
     }
-    $next_steps_links['admin/structure/menu/manage/main-menu'] = t('Manage the main menu');
-    $next_steps_links['admin/config/system/site-information'] = t('Update your site information');
-    $next_steps_links = $this->filterAvailableLinks($next_steps_links);
+    $node_types = node_type_get_types();
+    if (isset($node_types['post'])) {
+      $links_col2['node/add/post'] = t('Create a new @post_label', array('@post_label' => $node_types['post']->name));
+    }
+    $system_menus = menu_list_system_menus();
+    if (isset($system_menus['main-menu'])) {
+      $links_col2['admin/structure/menu/manage/main-menu'] = t('Update the %menu_name menu', array('%menu_name' => $system_menus['main-menu']));
+    }
+    $home_page_path = config_get('system.core', 'site_frontpage');
+    $home_layout_disabled = config_get('layout.layout.home', 'disabled');
+    if ($home_page_path == 'home' && !$home_layout_disabled) {
```

Comment:
Maybe you should also check that the layout for "home" exists in the first place? I might use `$home_layout = layout_load('home')` rather than directly reading config (layouts are cached with the API function).

Reply:
Good catch 👍

Reply:
resolved.